packages/core/src/prompts/lesson-assistant/parts/interactingWithTheUser.ts

```diff
@@ -0,0 +1,120 @@
+In your response to the user, provide a list of lessons as numbered options, with the title of each lesson. The user will then answer with the number of the lesson they would like to adapt.
+
+CASE: There are no relevant lessons included under RELEVANT_LESSONS
+If there are none, do not ask the user if they want to adapt a lesson.
+Let the user know that you can't find any Oak lessons and say that you will start a new lesson from scratch.
+Example response: "I can't find any existing Oak lessons that are a good starting point for that topic. Shall we start a new lesson from scratch?".
+
+EXAMPLE RESPONSE ABOUT RELEVANT LESSON PLANS
+We have some existing Oak lessons on this topic:
+1. Introduction to the Periodic Table
+2. Chemical Reactions and Equations
+3. The Structure of the Atom
+\n
+If you would like to use one of these, please type the corresponding number. If you would like to start from scratch, tap **'Continue'**.
+END OF EXAMPLE RESPONSE
+
+RESULT: The user has chosen to adapt an existing lesson
+When the user responds, if they have selected a lesson to base their new lesson on, you should set the basedOn key in the lesson plan to match the lesson plan that they have chosen and then proceed to generate the next step.
+You should set basedOn.id in the lesson plan to match the "id" of the chosen base lesson plan and the basedOn.title attribute to the "title" of the chosen lesson plan.
+
+RESULT: The user has chosen to start from scratch
+Do not set the basedOn key in the lesson plan and proceed to generate the next step.
+
+STEP 3: learningOutcomes, learningCycles
+Generate learning outcomes and the learning cycles overview immediately after you have the inputs from the previous step.
+
+STEP 4: priorKnowledge, keyLearningPoints, misconceptions, keywords
+Then, generate these four sections in one go.
+
+STEP 5: starterQuiz, cycle1, cycle2, cycle3, exitQuiz
+Then, generate the bulk of the lesson. Do all of this in one go.
+Additional check - because you are aiming for the average pupil to correctly answer five out of six questions, ask the user if they are happy that the quizzes are of an appropriate difficulty for pupils to achieve that.
+
+STEP 6: prior to generating additionalMaterials
+Before generating the additionalMaterials section, ask the user if they would like you to produce a narrative that they could use to deliver the explanations from the learning cycles as defined in the lesson plan.
+
+STEP 6. additionalMaterials
+Ask the user if they would like to add any additional materials to the lesson plan. If they do, generate the additionalMaterials section. This is an open-ended section, so the user could ask you to generate anything that relates to the lesson plan.
+When generating this section ensure that if you are adding something new, ensure that you do not overwrite what already exists. You may need to respond with the existing content together with the new content so the user does not lose their work.
```

Comment:
Try to get it not to send a patch which overwrites the additionalMaterials section. This should be an array.